sse/event_source.go

```diff
@@ -38,23 +38,47 @@ type EventSource struct {
 
 	currentReadCloser *ReadCloser
 	lastEventID       string
-	retryInterval     time.Duration
 	lock              sync.Mutex
 
-	closeOnce *sync.Once
-	closed    chan struct{}
+	closeOnce   *sync.Once
+	closed      chan struct{}
+	retryParams RetryParams
+}
+
+type RetryParams struct {
+	RetryInterval time.Duration
+	MaxRetries    uint16
 }
 
 func NewEventSource(client *http.Client, defaultRetryInterval time.Duration, requestCreator func() *http.Request) *EventSource {
+	retryParams := RetryParams{
+		RetryInterval: defaultRetryInterval,
+	}
+	return createEventSource(client, retryParams, requestCreator)
+}
+
+func createEventSource(client *http.Client, retryParams RetryParams, requestCreator func() *http.Request) *EventSource {
 	return &EventSource{
 		client:        client,
 		createRequest: requestCreator,
 
-		retryInterval: defaultRetryInterval,
+		closeOnce:   new(sync.Once),
+		closed:      make(chan struct{}),
+		retryParams: retryParams,
+	}
+}
+
+func ConnectWithParams(client *http.Client, retryParams RetryParams, requestCreator func() *http.Request) (*EventSource, error) {
```

Comment:
So do you mean to say that we create a `Config` struct something like this:
```
type Config struct {
	client         *http.Client
	retryParams    RetryParams
	requestCreator func() *http.Request
}
```
and add a `Dial` method to this struct that does what `ConnectWithParams` was doing and return `EventSource`? This means users of this library need to create an instance of `Config` struct and then invoke `Dial` to get `EventSource`. Correct?